app/containers/App/reducer.js

```diff
@@ -24,31 +23,26 @@ const initialState = fromJS({
 
 const appReducer = (state = initialState, action) => {
   switch (action.type) {
-    case SIGN_IN_SUCCESS:
-      const { token } = action;
-
-      return state
-        .set('token', token)
-        .setIn(['loggedUser', 'id'], parseJwt(token).id);
+    case STORE_TOKEN:
+      return state.set('token', action.token || '');
+    case STORE_USER_ID:
+      return state.setIn(['loggedUser', 'id'], action.id);
     case SIGN_IN_FAILURE:
       return state.set('token', '');
     case SHOW_LOADER:
       return state.set('isLoading', true);
     case HIDE_LOADER:
       return state.set('isLoading', false);
+    case GET_USER_SUCCESS: {
+      const userMeta = omit(action.user, 'id');
+
+      return state.setIn(['loggedUser', 'meta'], userMeta);
```

Comment:
You will fetch users from the api in other cases too, but it doesn't mean you have to store every user as the "logged" user. Please fix this.